assets/javascript/app.js

```diff
@@ -1,348 +0,0 @@
-                var infoDiv = $("<div>");
-                infoDiv.addClass("infobox-aside col-md-4");
-                var infoList = $("<ul>");
-
-                infoboxSplit.forEach(function(element) {
-                    if (element.indexOf("=") !== -1) {
-                        var pair = element.split("=");
-                        if ((wikiInfoBox.indexOf(pair[0].trim()) !== -1) && (pair[1].trim() !== "")) {
-                            var key = pair[0].trim();
-                            key = key.charAt(0).toUpperCase() + key.slice(1);
-                            var quant = pair[1].trim();
-                            quant = quant.replace(/\'\'/g, "\"");
-                            quant = quant.replace(/}}/g, "");
-
-                            infoList.append("<li>" + key + ": " + quant + "</li>");
-                        }
-                    }
-                });
-                infoDiv.append(infoList);
-                extractDiv.addClass("col-md-8");
-                contentDiv.append(extractDiv);
-                contentDiv.append(infoDiv);
-            }
-            else {
-                extractDiv.addClass("col-md-12");
-                contentDiv.append(extractDiv);
-            }
-            divMatches.append(contentDiv);
-        }
-
-        countAsync--;
-        checkAsyncResults();
-
-    }).fail(function(err) {
-        throw err;
-    });
-}
-
-
-
-function queryWikiAPI(searchWordArray) {
-
-    var wikiURL = "https://en.wikipedia.org/w/api.php?" + $.param({
-        "action": "query",
-        "list": "search",
-        "srsearch": searchWordArray.join(" "),
-        "srwhat": "text",
-        "format": "json"
-    });
-
-    $.ajax({
-        url: wikiURL,
-        dataType: "jsonp"
-    }).done(function(result) {
-        console.log("wikipedia query results", result.query.search);
-        countAsync = 0;
-        divMatches.empty();
-        result.query.search.forEach(function(article) {
-            if ((article.title.search("List")) && (article.title.search("Cloud")) && (article.title.indexOf("Media") === -1)) {
-                countAsync++;
-                // Need two API calls because can't return categories and extracts for each article in same call - generator max 1
-                parseWikiAPI(article.pageid, article.title);
-            }
-        });
-
-    }).fail(function(err) {
-        throw err;
-    });
-}
-
-
-
-function queryVisionAPI(imageData) {
-    // imageData is either a url or base 64 encoded string
-    var imageDataString;
-    if (isURL(imageData)) {
-        imageDataString = '"source":{"imageUri":"' + imageData + '"}';
-    } else {
-        imageDataString = '"content":"' + imageData + '"';
-    }
-    var request = '{"requests":[{"image":{' + imageDataString + '},"features":[{"type":"WEB_DETECTION","maxResults":10},{"type":"LABEL_DETECTION","maxResults":10}]}]}';
-
-    var response = new XMLHttpRequest;
```

Comment:
What is the reasoning behind using XMLHttpRequest instead of jQuery? There isn't anything wrong with this. However, some consistency is nice and jQuery is really nice because it supports multiple browsers as well as older versions of these browsers.